source/manual/test-and-build-a-project-with-github-actions.html.md

```diff
@@ -0,0 +1,266 @@
+management.
+
+```yml
+name: Continuous Integration
+on: [push, pull_request]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:9.4
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: --health-cmd "redis-cli ping" --health-interval 10s --health-timeout 5s --health-retries 5
+    env:
+      RAILS_ENV: test
+      REDIS_URL: redis://localhost:6379/0
+      TEST_DATABASE_URL: postgresql://postgres@localhost/content-publisher
+      GOVUK_CONTENT_SCHEMAS_PATH: vendor/govuk-content-schemas
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: Clone GOV.UK Content Schemas
+        uses: actions/checkout@v2
+        with:
+          repository: alphagov/govuk-content-schemas
+          ref: deployed-to-production
+          path: vendor/govuk-content-schemas
+      - uses: ruby/setup-ruby@v1
+      - name: Check for cached bundler
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: bundle-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: bundle
+      - run: bundle install --jobs 4 --retry 3 --deployment
+      - name: Install Node for JS dependencies
+        uses: actions/setup-node@v1
+      - run: npm install -g yarn
+      - name: Check for cached node modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: yarn
+      - run: yarn install --frozen-lockfile
+      - run: bundle exec rails db:setup
+      - run: bundle exec rails assets:precompile
+      - run: bundle exec rake
+```
+
+Notes:
+
+- Similar configuration to PostgreSQL and Redis can be applied to other software
+  dependencies such as MySQL, MongoDB or Rabbit MQ.
+- The `health-cmd` checks are used on services to ensure the build does not
+  continue until the service is running.
+- Additional repositories are cloned to the vendor directory for a consistent
+  location for third party code, this is consistent with Bundler.
+- Since the same actions are used in different contexts the `name` option has
+  been used in steps to distinguish the different steps.
+- The `bundle exec rails assets:precompile` task is used to build a warm cache
+  of assets before starting tests to reduce risk of timeouts, this is due to
+  an [underlying performance issue][govuk-frontend-issue] with GOV.UK asset
+  compilation.
+- In the underlying project (Content Publisher) `bundle exec rake` performs:
+  Ruby & JS unit testing; Ruby, SCSS, JavaScript and FactoryBot linting; and
+  brakeman security audit.
+- This example does not include tagging the [master branch with a particular
```

Comment:
Should it? Or does this have bearing on our "no secrets" decision? Though it should use the `GITHUB_TOKEN` from the Action run, not one of our own...

Reply:
Yeah the GITHUB_TOKEN will have access. It'd probably be useful to have an example but that might not be applicable until we have one to use in anger.

I'm not too sure at this point whether we'd continue with incrementing integer or use an approach based on sha hash?